account-kit/smart-contracts/src/light-account/accounts/viem-multiowner.ts

```diff
@@ -0,0 +1,134 @@
+import type { SmartAccountSigner } from "@aa-sdk/core";
+import {
+  encodeFunctionData,
+  type Address,
+  type Chain,
+  type Hex,
+  type PublicClient,
+  type Transport,
+} from "viem";
+import { readContract } from "viem/actions";
+import { MultiOwnerLightAccountAbi } from "../abis/MultiOwnerLightAccountAbi.js";
+import { MultiOwnerLightAccountFactoryAbi } from "../abis/MultiOwnerLightAccountFactoryAbi.js";
+import { getDefaultMultiOwnerLightAccountFactoryAddress } from "../utils.js";
+import { predictMultiOwnerLightAccountAddress } from "./predictAddress.js";
+import {
+  createViemLightAccountBase,
+  type ViemLightAccountBase,
+} from "./viem-base.js";
+
+export type ViemMultiOwnerLightAccount<
+  TSigner extends SmartAccountSigner = SmartAccountSigner,
+> = ViemLightAccountBase<TSigner, "MultiOwnerLightAccount", "v2.0.0"> & {
+  encodeAddOwner: (ownerToAdd: Address) => Hex;
+  encodeRemoveOwner: (ownerToRemove: Address) => Hex;
+  getOwnerAddresses: () => Promise<readonly Address[]>;
+};
+
+export type CreateViemMultiOwnerLightAccountParams<
+  TTransport extends Transport = Transport,
+  TSigner extends SmartAccountSigner = SmartAccountSigner,
+> = {
+  client: PublicClient<TTransport, Chain>;
+  signer: TSigner;
+  owners: Address[];
+  salt?: bigint;
+  accountAddress?: Address;
+  factoryAddress?: Address;
+};
+
+export async function createViemMultiOwnerLightAccount<
+  TTransport extends Transport = Transport,
+  TSigner extends SmartAccountSigner = SmartAccountSigner,
+>({
+  client,
+  signer,
+  owners,
+  salt = 0n,
+  accountAddress,
+  factoryAddress,
+}: CreateViemMultiOwnerLightAccountParams<TTransport, TSigner>): Promise<
+  ViemMultiOwnerLightAccount<TSigner>
+> {
+  const signerAddress = await signer.getAddress();
+
+  // Deduplicate and sort owners
+  const deduplicatedOwners = Array.from(
+    new Set([signerAddress, ...owners].map((owner) => owner.toLowerCase())),
+  ).map((owner) => owner as Address);
```

Comment:
✨ this final `map` seems kinda useless? not the biggest deal though since the array will be small.
```suggestion
  const deduplicatedOwners = Array.from(
    new Set([signerAddress, ...owners].map((owner) => owner.toLowerCase())),
  ) as Address[];
```

Reply:
yea you're right